pkgs/by-name/ni/niri/package.nix

```diff
@@ -89,6 +89,10 @@ rustPlatform.buildRustPackage rec {
     + lib.optionalString withSystemd ''
       install -Dm0755 resources/niri-session -t $out/bin
       install -Dm0644 resources/niri{-shutdown.target,.service} -t $out/lib/systemd/user
+    ''
+    + lib.optionalString withDinit ''
+      install -Dm0755 resources/niri-session -t $out/bin
```

Comment:
if `withSystemd = true` and `withDinit = true`, then this will be installed twice. It doesn't really matter, since it'll just be overwritten with an identical file (and this case is probably quite rare anyway, you'd generally want to build with just one, though they are not mutually exclusive). [maybe merge them?](https://github.com/sodiboo/niri-flake/blob/3dafb3bf5e23ee2aa3f24aad6bff300df7311558/flake.nix#L161-L164)

Reply:
Good point, will do so tomorrow!